nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -1412,6 +1443,82 @@ async fn save_sp_dumps(
     Ok(())
 }
 
+/// Write a file with a JSON mapping of sled serial numbers to cubby and UUIDs for easier
+/// identification of sleds present in a bundle.
+async fn write_sled_info(
+    log: &slog::Logger,
+    mgs_client: &MgsClient,
+    nexus_sleds: Option<&[Sled]>,
+    dir: &Utf8Path,
+) -> anyhow::Result<()> {
+    #[derive(Serialize)]
+    struct SledInfo {
+        cubby: Option<u16>,
+        uuid: Option<Uuid>,
+    }
+
+    let available_sps = get_available_sps(&mgs_client)
+        .await
+        .context("failed to get available SPs")?;
+
+    // We can still get a useful mapping of cubby to serial using just the data from MGS.
+    let mut nexus_map: BTreeMap<_, _> = nexus_sleds
+        .unwrap_or_default()
+        .into_iter()
+        .map(|sled| (sled.serial_number(), sled))
+        .collect();
+
+    let mut sled_info = BTreeMap::new();
+    for sp in
+        available_sps.into_iter().filter(|sp| matches!(sp.type_, SpType::Sled))
+    {
+        let sp_state = match mgs_client.sp_get(&sp.type_, sp.slot).await {
+            Ok(s) => s.into_inner(),
+            Err(e) => {
+                error!(log,
+                    "Failed to get SP state for sled_info.json";
+                    "cubby" => sp.slot,
+                    "component" => %sp.type_,
+                    "error" => InlineErrorChain::new(&e)
+                );
+                continue;
+            }
+        };
+
+        if let Some(sled) = nexus_map.remove(sp_state.serial_number.as_str()) {
+            sled_info.insert(
+                sp_state.serial_number.to_string(),
+                SledInfo {
+                    cubby: Some(sp.slot),
+                    uuid: Some(*sled.identity.id.as_untyped_uuid()),
+                },
+            );
+        } else {
+            sled_info.insert(
+                sp_state.serial_number.to_string(),
+                SledInfo { cubby: Some(sp.slot), uuid: None },
+            );
+        }
+    }
+
+    // Sleds not returned by MGS.
+    for (serial, sled) in nexus_map {
+        sled_info.insert(
+            serial.to_string(),
+            SledInfo {
+                cubby: None,
+                uuid: Some(*sled.identity.id.as_untyped_uuid()),
+            },
+        );
```

Comment:
Smart to still include this info if a sled is present in one of "the nexus list" and "the MGS cubby list" but not the other